packages/kbn-tinymath/grammar/grammar.peggy

```diff
@@ -11,6 +11,32 @@
       max: location.end.offset
     }
   }
+
+  const symbolsToFn = {
+    '+': 'add', '-': 'subtract',
+    '*': 'multiply', '/': 'divide',
+    '<': 'lt', '>': 'gt', '==': 'eq',
```

Comment:
any reason we aren't going with `=` for equality? As we don't have assignment, it seems like we can go with the simpler case which is also what excel is doing.

Reply:
Named parameters already use the `=` syntax ( i.e. `count(kql="...")`).
It should work already, but thought that the `==` operation is something quite familiar as well and different enough to not confuse the two.

Reply:
it's familiar for developers, I don't think it's super common outside of that group. SQL also doesn't do it. What do you think @ghudgins @ninoslavmiskovic ?

Reply:
I also believe that "==" is not known among no developers, and if SQL does not support it, then it is also an indicator that it would be the preferable choice since SQL is a broader query language than KQL. Are there any cases where it will be an issue of sticking with "=" ?

Reply:
Made a quick spike on that to see whether it is possible to co-exists with the named argument.
Unfortunately there are some overlaps between the two which make it harder to solve it, at least at grammar level:

```
ifelse(a=1, 1, 2) // => 🆘  named argument or comparison? Grammar says it's named argument
ifelse(a=a, 1, 2) // => 🆘  named argument or comparison? Grammar says it's named argument
ifelse(1=1, 1, 2) // => ✅ comparison
ifelse(1=a, 1, 2) // => ✅ comparison
ifelse(unsupported-named-argument=1, 1, 2) // => ✅ comparison
```

Reply:
It looks like it should parse correctly, but I didn't check

Reply:
They all parse fine.
As long as there's no future plan to have string comparison in the future it can work.

Reply:
OK, in that case I think we should go with the single `=`. Otherwise this looks good to me!

Reply:
Had some more investigation on the topic and found out some more issues on the usage for the single `=` symbol for comparison.
Here some examples:
* ❌  for plain tinymath it is possible to use a variable for the comparison on `exec`
* ❌ in `Canvas` it is possible to use a variable for the comparison: `math "ifelse(total_errors = 37, 1, 0)"` which will raise an error about unsupported `Named parameters`. This is using `exec` underneath. This is probably a non-issue as conditional logic can be performed elsewhere 
* ❌ If in the future an optimization like #141456 is introduced for comparison, this can be a problem also for Lens in the formula rewrite step.

All of them are minor problems, but still to keep in mind if we decide to go down this single `=` symbol route rather than the existing `==`.

Reply:
Discussed offline and decided to keep the `==` symbol.